runtime/gateway.go

```diff
@@ -800,6 +803,7 @@ func (gateway *Gateway) setupTChannel(config *StaticConfig) error {
 	if !ok {
 		return errors.Errorf("unknown sub logger level for tchannel server: %s", levelString)
 	}
+	gateway.TChannelSubLoggerLevel = level
 
```

Comment:
Required in case `dedicated.tchannel.client: true` in which case we need this config while creating clients in `init phase`

Reply:
qq: How is logger level related to TChannel?

Reply:
- When we create a new TChannel, we need to specify a `tchannel.Logger`. 
- To pass this, we create a new TChannelLogger value. 
- To pass this value, we create a sublogger based on `gateway.logger` which essentially clones the original logger. 
- The level we pass to this logger `debug|info|...` is determined by `gateway. TChannelSubLoggerLevel`
- `gateway.TChannelSubLoggerLevel` is populated by `subLoggerLevel.tchannel`

```
func NewTChannelLogger(logger *zap.Logger) tchannel.Logger {}
func (gateway *Gateway) SubLogger(name string, level zapcore.Level) *zap.Logger {}
```